### PR TITLE
[feature] Added a system event to allow developers to assign Smarty variables

### PR DESCRIPTION
**Pull request by rtripault**
### What does it do ?

Adds a system event to enable developers to assign Smarty variables to be used in user navigation.
### Why is it needed ?

Smarty variables could be used, ie., to display a notification count number on a menu entry.
### To do before merge
- [ ] triple check it is not possible to override critical variables (ie. `$_config`)
- [ ] make sure Smarty variables are usable/parsed in lexicon strings

### CODE REVIEW

**Comment by Mark-H** [comment]
Any update on those to dos? Looks like it could be a cool feature.

**Comment by rtripault** [comment]
Nope. I guess i was looking for some kind of feedback before digging in those :)

**Comment by Mark-H** [comment]
Well the use case you mentioned (notification counts) is really appealing and I can already think of a few extras I could consider adding it too (Scheduler with the number of scheduled runs, SimpleCart with the number of new orders etc). 

Thinking about it some more I'm not sure if adding "global variables" to smarty is the best approach to dealing with something like that.. maybe (from an extras dev perspective) adding a simple `<span class="scheduler-run-count"></span>` to the menu lexicon and using Ext.get().update (maybe with some ajax) would be a bit cleaner? It then also doesn't restrict you to using that lexicon in smarty, the same lexicon key from the menu could also be used for a component header or something, which is js/ext/`_`..

**Comment by rtripault** [comment]
You are definitely right, the use case (counts) would be best suited/filled with some ajax. And since i don't really remind the other awesome use cases i could have thought about when submitting the PR (and the possible drawbacks to it), i'd be up to close it
